### DISCUSSION TITLE
Can clangd also get the built-in defines using the query-driver?

### DISCUSSION THREAD

**Discussion by itavero** [clangd/clangd#2109] [Q&A]
I'm using clangd (18.1.8) to analyze sources that are being cross-compiled (for instance with arm-none-eabi-gcc, but also other GCC based compilers).
By specifying the query driver, it did manage to find the system include headers.

Unfortunately, now it complains that `__WCHAR_MIN__` is not declared. This is however something that the compiler defines (e.g. if I run `arm-none-eabi-gcc -dM -E - < /dev/null > gcc_macros.h` it will be in that file).
Can I somehow ask clangd to do this step as well and take over the defines it does not know about yet?

The warning I get is:
```
In included file: use of undeclared identifier '__WCHAR_MIN__' [undeclared_var_use]
```

**Comment by HighCommander4** [comment]
Duplicate of https://github.com/clangd/clangd/issues/533

**Reply by itavero** [reply]
Good catch. I see some potential solutions there that I can probably automate.
